File: src/Card.py
class Card:

    # colour = 1 - heart
    # colour = 2 - diamond
    # colour = 3 - clubs
    # colour = 4 - spades
    def __init__(self, name, colour, value):
        self.name = name
        self.colour = colour
        self.value = value

    def getName(self):
        return self.name

    def getValue(self):
        return self.value

    def getColour(self):
        return {
            1: 'heart',
            2: 'diamond',
            3: 'club',
            4: 'spade'
        }[self.colour]

File: src/org/foi/jurbunic/bela/cards/Card.java
package org.foi.jurbunic.bela.cards;

import java.io.Serializable;

public class Card implements Serializable {

    private String name;
    private String shortName;
    private Colour colour;
    private double value;

    public Card(String name, String shortName, Colour colour, double value) {
        this.name = name;
        this.shortName = shortName;
        this.colour = colour;
        this.value = value;
    }


    public String getName() {
        return name;
    }

    public Colour getColour() {
        return colour;
    }

    public String getColourName(){
        return colour.getColourName();
    }
    public double getValue() {
        if(this.getColour().isTrump()){
            if(this.getName().equals("Jack")){
                return 20;
            }
            if(this.getName().equals("Nine")){
                return 14;
            }
        }
        return value;
    }

    public String getShortName() {
        return shortName;
    }

    public String getCardGraphic(){

        return "|"+getShortName()+getColour().getColourSymbol()+"| ";
    }
}
